TACC/Frontera-Portal
# Task/fp 639 message component updates

## To Do: ##

- [ ] From Design, get inverted warning icon or create story to update.
- [ ] With Design, confirm text color and icon for errors or create story to update.

## Overview: ##

For Messages
- support scopes
- update styles
- support visibility toggle
- support dismissal callback
- support wrapper components (for convenience)

## PR Status: ##


## Related Jira tickets: ##

* [FP-639](https://jira.tacc.utexas.edu/browse/FP-639)
    * [FP-640](https://jira.tacc.utexas.edu/browse/FP-640)

## Summary of Changes: ##

- __New__: Update `UIPatternsMessages` to showcase all `Message` features
- __New__: Create `_common/InlineMessage` component
- __New__: Update `_common/Message` component
    - new styles
    - new features
    - polish scripts
- __New__: Create `_common/SectionMessage` component
- __New__: Update global color variables

## Testing Steps: ##

### Test Patterns
1. Load "UI Patterns" page.
1. Ensure messages with X button can be dismissed.
1. Ensure inline messages do __not__ have an X button.
1. Ensure styles differ and match consistently across the matrix.

### Test Existing Code
1. Load "Applications".
1. Open app to which you have no access (perhaps "My Apps" → "Frontera HPC COVID-19 Analysis Portal").
1. See inline warning message.
1. See warning in console to set type or use wrapper component.

### Apply to Existing Code
1. With branch checked out, `git cherry-pick 0e95409d341dd95d5f3f7fdae000c318b938d7f3`.
1. With no allocation, load "Applications".

## UI Photos:

![FP-639 Patterns](https://user-images.githubusercontent.com/62723358/99760830-4384ea80-2aba-11eb-92f0-3f4ff11c5e68.png)

## Notes: ##

The design is a dev-driven, design-tweaked amalgamation of styles. The styles may improve in the future, but a component makes that change automatically applied across the app.
